kore/src/Kore/Equation/Application.hs

```diff
@@ -516,6 +594,8 @@ data ApplyMatchResultError variable
     -- term was required.
     | NotMatched (SomeVariableName variable)
     -- ^ The variable was not matched.
+    | NonTargetSubstitution (SomeVariableName variable)
```

Comment:
I think we should call this `NonMatchingSubstitution` instead because having this substitution means that the solution is not a matching solution (and the documentation should say that).